File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0
def dfs(numbers,val,target, N, size):
    global cnt
    #print(val)
    if N==size-1:
        if val==target:
            cnt+=1
            #print(cnt)
        return cnt

    dfs(numbers,val-numbers[N+1],target, N+1, size)
    dfs(numbers,val+numbers[N+1],target, N+1, size)
    
    return cnt
    
def solution(numbers, target):
    global cnt
    answer = 0
    answer=dfs(numbers,numbers[0],target, 0, len(numbers))
    answer=dfs(numbers,-numbers[0],target, 0, len(numbers))
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    static int dfs(int val, int idx, int[] numbers, int target) {
        if (idx==numbers.length-1) {
            // System.out.println(val);
            if (val==target) {
                answer++;
            }
            return val;
        }
        
        dfs(val+numbers[idx+1], idx+1, numbers, target);
        dfs(val-numbers[idx+1], idx+1, numbers, target);
        
        return val;
    }
    
    public int solution(int[] numbers, int target) {
        
        dfs(numbers[0], 0, numbers, target);
        dfs(-numbers[0], 0, numbers, target);
        
        return answer;
    }
}